library/alloc/src/rc.rs

```diff
@@ -2843,6 +2845,7 @@ impl<T: ?Sized, A: Allocator> Weak<T, A> {
     /// [`null`]: ptr::null
     #[must_use]
     #[stable(feature = "rc_as_ptr", since = "1.45.0")]
+    #[cfg_attr(not(bootstrap), rustc_never_returns_null_ptr)]
```

Comment:
The documentation indicates that the pointer can be null:
> The pointer is valid only if there are some strong references. The pointer may be dangling,
> unaligned or even [`null`] otherwise.

but the code indicates otherwise as `Weak` internally uses `NonNull` :thinking: